src/build/arg/mod.rs

```diff
@@ -4043,6 +4047,34 @@ impl<'help> Arg<'help> {
         self
     }
 
+    /// Sets a hint about the type of the value for shell completions
+    ///
+    /// Currently this is only supported by the zsh completions generator.
+    ///
+    /// For example, to take a username as argument:
+    /// ```
+    /// # use clap::{Arg, ValueHint};
+    /// Arg::with_name("user")
+    ///     .short('u')
+    ///     .long("user")
+    ///     .takes_value(true)
+    ///     .value_hint(ValueHint::Username)
+    /// # ;
+    /// ```
+    ///
+    /// To take a full command line and its arguments:
+    /// ```
+    /// # use clap::{Arg, ValueHint};
+    /// Arg::with_name("command")
+    ///     .multiple(true)
+    ///     .value_hint(ValueHint::CommandWithArguments)
+    /// # ;
+    /// ```
+    pub fn value_hint(mut self, value_hint: ValueHint) -> Self {
```

Comment:
Also, can there only be one value hint, can't it be a merged value hint or a list?

Reply:
Hmm, can you imagine any use cases for combining multiple hints for one argument?

Seems it would be quite an edge case. For situations like that, I think it would be better to provide some "escape hatch" to manually override parts of the generated completion file, rather than trying to accommodate such weirdness.